File: 0020-0029/0025.py
import sys

lists = []
for str in sys.stdin:
  lists+=[tuple(str.rstrip().split(' '))]

for i in range(0, len(lists), 2):
  hit = brow = 0
  for j in range(4):
    if lists[i][j] == lists[i+1][j]:
      hit += 1
  for c in lists[i]:
    if c in lists[i+1]:
      brow += 1
  print "%d %d"%(hit, brow-hit)

File: 0020-0029/0025.rb
while gets do
  ar = $_.split(' ').map(&:to_i)
  br = gets.split(' ').map(&:to_i)
  hit = ar.map.with_index { |a, i| a == br[i] ? 1 : 0 }.inject(&:+)
  brow = (ar&br).size - hit
  puts "#{hit} #{brow}"
end